Sources/TuistGraph/Models/CompatibleXcodeVersions.swift

```diff
@@ -1,20 +1,70 @@
 import Foundation
+import struct ProjectDescription.Version
 
 /// Enum that represents all the Xcode versions that a project or set of projects is compatible with.
-public enum CompatibleXcodeVersions: Equatable, Hashable, ExpressibleByArrayLiteral {
+public enum CompatibleXcodeVersions: Equatable, Hashable, ExpressibleByArrayLiteral, ExpressibleByStringInterpolation,
+    CustomStringConvertible
+{
     /// The project supports all Xcode versions.
     case all
 
+    case exact(Version)
+
+    case upToNextMajor(Version)
+
+    case upToNextMinor(Version)
+
     /// List of versions that are supported by the project.
-    case list([String])
+    case list([CompatibleXcodeVersions])
+
+    public func isCompatible(versionString: String) -> Bool {
+        let xCodeVersion: Version = "\(versionString)"
+
+        switch self {
+        case .all:
+            return true
+        case let .exact(version):
+            return version == xCodeVersion
+        case let .upToNextMajor(version):
+            return version == xCodeVersion || (version.major == xCodeVersion.major && xCodeVersion > version)
```

Comment:
Shouldn't be `>=`? Also, let's try to always put the variable as first operand of the comparison. Same for the upToNextMinor case
```suggestion
            return xCodeVersion == version || (xCodeVersion.major == version.major  && xCodeVersion >= version)
```

Reply:
If they're equal it will short-circuit due to the first clause, so that part won't be executed.

Reply:
True, I would rather remove the first == to make the logic clearer